src/Misc/PerformanceViewer/performanceViewerCollector.ts

```diff
@@ -478,8 +478,10 @@ export class PerformanceViewerCollector {
         if (!shouldPreserve) {
             this.datasets.data = new DynamicFloat32Array(initialArraySize);
             this.datasets.startingIndices = new DynamicFloat32Array(initialArraySize);
+            this._startingTimestamp = PrecisionDate.Now;
+        } else if (!this._startingTimestamp) {

```

Comment:
did you mean to check if it is undefined, or is any false-y value good here?

Reply:
Oh, it should be undefined, thanks for catching that!